### PR TITLE
Sync excludes collections found at /excludes/ endpoint

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #8194

### CODE REVIEW

**Review by rochacbruno — APPROVED** [review]

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -455,6 +476,9 @@ async def _add_collection_version(self, api_version, collection_version_url, met
             version=metadata["version"],
         )
         cv_unique = attrgetter("namespace", "name", "version")(collection_version)
+        fullname = ".".join(cv_unique[:2])
+        if fullname in self.exclude_info and cv_unique[2] in self.exclude_info[fullname]:
```
Comment (fao89) [inline pulp_ansible/app/tasks/collections.py:480]: could you add a test for checking the scenario when we want to exclude only a specific version?

File: pulp_ansible/app/tasks/collections.py [hunk 2]
```diff
@@ -634,15 +645,32 @@ async def _download_unpaginated_metadata(self):
...
-            try:
-                collection_metadata_list = parse_metadata(await downloader.run())
-            except FileNotFoundError:
-                pass
-            else:
+            exc_downloader = self.remote.get_downloader(
+                url=excludes_endpoint, silence_errors_for_response_status_codes={404}
+            )
+            tasks = [loop.create_task(col_downloader.run()), loop.create_task(exc_downloader.run())]
+            results = await asyncio.gather(*tasks, return_exceptions=True)
```
Comment (fao89) [inline pulp_ansible/app/tasks/collections.py:659]:
```suggestion
            results = await asyncio.gather(*tasks, return_exceptions=True)
            col_result, exc_result = results
```
and then replace results[0]...

File: pulp_ansible/app/tasks/collections.py (hunk 1 above)
Comment (fao89) [reply pulp_ansible/app/tasks/collections.py:480]:
```suggestion
        fullname = ".".join(cv_unique[:2])
        if fullname in self.exclude_info and collection_version.version in self.exclude_info[fullname]:
```
`cv_unique[2]` made me confuse, I had to read multiple times to understand

**Comment by mdellweg** [comment]
Should we backport to all versions starting from 0.7?

**Pull request merged**